kernel/src/platform/mpu.rs

```diff
@@ -263,8 +268,24 @@ pub trait MPU {
     fn configure_mpu(&self, config: &Self::MpuConfig);
 }
 
-/// Implement default MPU trait for unit.
-impl MPU for () {
+/// No-op MPU implementation, providing no isolation guarantees.
+///
+/// Using this implementation violates Tock's isolation guarantees and results
+/// in applications having unrestricted access to kernel memory. As such,
+/// constructing this type is an `unsafe` operation.
+// By having this type contain a private field, we prevent constructing it from
+// outside this crate, except through the `unsafe fn new()` constructor.
+pub struct NopMPU(());
+
+impl NopMPU {
+    pub unsafe fn new() -> Self {
+        NopMPU(())
+    }
+}
+
+/// This type does not meet the safety requirements outlined on the `MPU` trait,
+/// hence constructing it is also an unsafe (and unsound) operation.
+unsafe impl MPU for NopMPU {
```

Comment:
I'm concerned that we cannot do this. I agree that making the constructor unsafe is a good companion, but I don't think it is valid to blatantly ignore the safety requirement.